### PR TITLE
Cumulative PR about infrastructure and code cleaning

### PR DESCRIPTION
**Pull request by Gesugao-san**
https://youtu.be/ZX8PgpML5AY

### CODE REVIEW

**Comment by RocketPuppy** [comment]
@Gesugao-san Sorry it took me a while to get to this. I skimmed it but won't be able to take a closer look at it for a few days. Some of this I'll keep and some of it I won't. You did a lot of work here and I appreciate the cleanups and restructuring. The self-injecting head/nav stuff isn't going to make it in unless you can make a convincing case for it. I want this to remain simple, with the only javascript being stuff that's required for tool functionality, and I don't understand what those self-injecting additions are adding in those terms. I'm going to also need to take a closer look at the new folder structure. There was a method to my madness in the original structure.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+charset = utf-8
+end_of_line = crlf
+files.encoding = "utf8"
+indent_size = 2
```
Comment (RocketPuppy) [inline .editorconfig:10]: I'm a four-space person, sorry. This will need to change to `4` instead of `2`.

File: resources/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+[*.svg]
+indent_size = 3
```
Comment (RocketPuppy) [inline resources/.editorconfig:4]: Four spaces please.

File: test-branch.html [hunk 3]
```diff
@@ -1,9 +1,17 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <script type="module" defer src="./js/armor-testing.js"></script>
-  <link rel="stylesheet" href="./css/test-branch.css" />
+  <script defer src="./js/head.js"></script>
```
Comment (RocketPuppy) [inline test-branch.html]: Why is this any better than just including the tags manually?

**Comment by RocketPuppy** [comment]
TLDR: I have cherry-picked commits 2a34d30, 1d1763e, and 5454acf. The latter two with some changes. You can remove those from this PR. The rest of the commits I have asked for more substantial changes, described in more detail below.

Review:
Commit [2a34d30](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/2a34d3051a2da6f5e5b794f0e587ffc7c123b057), this is good. I'll pull this in. Thank you.

Commit [1d1763e](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/1d1763ee30b8411c412836d924c4e49d66a682ca). Aside from the vscode stuff (which I'm happy to pull in), I'm not going to take this commit. The files are organized the way they are on purpose. All of the cross-cutting concerns are at the root level, and everything that's specific to a given page is in a subfolder for that page. That's intentional to help keep the knowledge required to work on a given page to a minimum. I just go to that folder and everything I need is there.

Commit [5454acf](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/5454acfd9a30f6dddeaec47bda415b0a6c6523a2) is good, thank you. But I'm going to skip the favicon. I don't want to have the same favicon as the wiki and cause people to be confused about which site they are on.

Commit [d8a99ef](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/d8a99efaa8b9352ea1d96b21c9a9fa6793fdd817). I don't mind including the EditorConfig file, but I left a comment about it's content. If you can address that it can be included.

Commit [574fa15](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/574fa155272f2e927ff1d969cd601e8f7394e8e0) is good work for the most part. I don't want to include the head.js code though. And there's another comment about indentation on it. If you can address that it can be included.

Commit [b118da8](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/b118da8b3071eed0c2ef50b9d528a1fbcd730357) I will not be including. I don't see any benefit to the self-injecting head over having the head tags in the markup.

Commit [4cc9dc0](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/4cc9dc0a505d0d9371302e9208f9289b6b5b3a8e) I don't think is necessary. The repo already deploys to Github pages just fine without a workflow defined.

Commit [b7d88e3](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/b7d88e366c2ee8356a9771e60dbfd598e5785f6b) I will not be including. As mentioned previously I see no benefit to the self-injecting code over just including the tags in the markup. There are a couple cosmetic changes in this commit that can be included if they are extracted from the self-injecting code.

Commit [327ee04](https://github.com/RocketPuppy/Nebulous-Fleet-Command-Minmax-Tool/pull/5/commits/327ee04885cd0cc35cf82e52823ce5b3c3f69cbe) relies a lot on commit 1d1763e, specifically the file structure changes, which I will not be including. However there are some other small changes in this commit that are worth including if you can remove the changes that rely on the file structure changes. I will also not be including the use of the NPM live-server package. I really don't like working with NPM and I'd rather not have it in my projects if I can help it. The better way to serve the files locally is to use python 3, with `python -m http.server` in the root directory. No need to install anything other than python, which is likely to already be installed as it is a common requirement.

**Comment by RocketPuppy** [comment]
has a ton of conflicts now and comments never addressed. Reopen if that changes.
